File: 프로그래머스/lv1/12934. 정수 제곱근 판별/정수 제곱근 판별.js
// 정수 제곱근 판별?
  // 반복문 돌려서 될때까지 하는건 너무 비효율..뭐더라
  // 유클리드 호제법? ㄴㄴ

// function solution(n) {
//   let answer = 0
//   let sqrtN = Math.sqrt(n)
//   Number.isInteger(sqrtN) === true ? answer = Math.pow(sqrtN+1,2) : answer = -1

//   return answer
// }
function solution(n) {

  let sqrtN = Math.sqrt(n)

  return Number(sqrtN) === parseInt(sqrtN) ? (sqrtN+1) ** 2 : -1
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.py
# def solution(n):
#     # 3항 연산자
#     sqrt = n**(1/2)
#     return (sqrt+1)**2 if type(sqrt)==int else -1 # 11.0은 int가 아니라 안됨

def solution(n):
    sqrt = n**(1/2)
    return (sqrt+1)**2 if (sqrt%1 == 0) else -1